### PR TITLE
fix.trusted_network_has_highest_priority.ag

### PR DESCRIPTION
**Pull request by alexandergull**

### CODE REVIEW

**Review by Glomberg — APPROVED** [review]

**Review by svfcode — APPROVED** [review]

File: uniforce/lib/Cleantalk/USP/Security/Firewall.php [hunk 1]
```diff
@@ -166,43 +166,58 @@ public function run() {
...
-		if( is_array( $results ) ) {
-			foreach ( $results as $fw_result ) {
-				$priority = array_search( $fw_result['status'], $this->statuses_priority ) + ( isset($fw_result['is_personal']) && $fw_result['is_personal'] ? count ( $this->statuses_priority ) : 0 );
-				if( $priority >= $current_fw_result_priority ){
-					$current_fw_result_priority = $priority;
+
+        $final_priority = 0;
+        $result = array(
+            'status' => 'PASS',
+            'ip' => '',
```
Comment (ArtemAnoshin) [inline uniforce/lib/Cleantalk/USP/Security/Firewall.php:173]: If the array does not come to the parameter, then the result with an empty value in the IP field is suitable for us?

File: uniforce/lib/Cleantalk/USP/Security/Firewall.php [hunk 2]
```diff
@@ -166,43 +166,58 @@ public function run() {
...
+
+        if ( is_array($results) ) {
+            foreach ( $results as $fw_result ) {
+                /**
+                 * Calculating priority. Records that have status PASS_BY_TRUSTED_NETWORK gain hardcoded highest priority.
+                 * Personal records are next by priority in accordance of this->statuses_priority.
+                 */
+                $status_priority_from_table = array_search($fw_result['status'], $this->statuses_priority);
+                $is_personal_flag = isset($fw_result['is_personal']) && $fw_result['is_personal'];
+                $is_trusted_network_flag = isset($fw_result['status']) && $fw_result['status'] == 'PASS_BY_TRUSTED_NETWORK';
```
Comment (ArtemAnoshin) [inline uniforce/lib/Cleantalk/USP/Security/Firewall.php:184]: Maybe if the status PASS_BY_TRUSTED_NETWORK is found, immediately return the result?

Comment (alexandergull) [reply uniforce/lib/Cleantalk/USP/Security/Firewall.php:184]: Prioritize logic is cloned from the WordPress plugin, I presume we would better keep this as is.

File: uniforce/lib/Cleantalk/USP/Security/Firewall.php (hunk 1 above)
Comment (alexandergull) [reply uniforce/lib/Cleantalk/USP/Security/Firewall.php:173]: Yes, if no module results are gained we just pass this request through. Note, this logic is used twice, for each module and for the conjunction of modules results.

**Pull request merged**